crawl-ref/source/godabil.cc

```diff
@@ -4242,6 +4242,49 @@ bool gozag_setup_potion_petition(bool quiet)
     return true;
 }
 
+// Would a certain potion effect be immediately useful?
+// TODO: check duration for all buff-extending effects (might, haste, etc)
+int _potion_effect_pricing(item_def potion)
+{
+    int price = item_value(potion, true);
+    switch (potion.sub_type)
```

Comment:
This switch statement is too big. I'd like to reduce the price of any potion if `!get_potion_effect(potion.sub_type)->can_quaff()` but that formulation gives me a type casting error. Not sure what I'm doing wrong.
Once that's done, only special cases will belong in this select statement (like flight & berserk).